### PR TITLE
[docs][APM] Update APM/Beats documentation

### PR DESCRIPTION
**Pull request by bmorelli25**
This PR persists changes made in the following apm-server PRs:

* APM Server install and run structure https://github.com/elastic/apm-server/pull/2170
* Update Logstash output documentation https://github.com/elastic/apm-server/pull/2691
* Add and improve SSL/TLS documentation https://github.com/elastic/apm-server/pull/2715

- [x] ~This PR can only be merged after Beats switches to Asciidoctor.~

### CODE REVIEW

File: libbeat/docs/command-reference.asciidoc [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 
 // These attributes are used to resolve short descriptions
+tag::attributes[]
```
Comment (bmorelli25) [inline libbeat/docs/command-reference.asciidoc]: These are awesome!

File: libbeat/docs/command-reference.asciidoc [hunk 2]
```diff
@@ -723,15 +732,27 @@ ifndef::no_dashboards[]
 {beatname_lc} setup --index-management
 -----
 endif::no_dashboards[]
+
+ifndef::apm-server[]
```
Comment (bmorelli25) [inline libbeat/docs/command-reference.asciidoc]: Unfortunately I had to nest conditionals here. If you can think of a better idea, let me know.

File: libbeat/docs/outputconfig.asciidoc [hunk 3]
```diff
@@ -739,21 +739,39 @@ use in Logstash for indexing and filtering:
...
 <<logstash-index,`index`>> option in the {beatname_uc} config file.
-<3> The beats current version.
-<4> The value of `type` is currently hardcoded to `doc`. It was used by previous
-Logstash configs to set the type of the document in Elasticsearch.
-
+<3> The current version of {beatname_uc}.
+endif::[]
 
-WARNING: The `@metadata.type` field, added by the Logstash output, is
-deprecated, hardcoded to `doc`, and will be removed in {beatname_uc} 7.0.
```
Comment (bmorelli25) [inline libbeat/docs/outputconfig.asciidoc]: This has been deprecated for a very long time and was not removed in `7.0` like the note says. I opted to remove this note and the `type` from the metadata. If you prefer to keep it around, let me know and I'll change this.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
Woo, ci/docs is green on this one! Ready for a review when you have a moment @dedemorton.

The [diff](http://beats_13596.docs-preview.app.elstc.co/diff) looks good to me -- all changes are expected.

File: libbeat/docs/command-reference.asciidoc [hunk 4]
```diff
@@ -646,11 +649,12 @@ ifdef::has_ml_jobs[]
 necessary to analyze data for anomalies.
 endif::[]
 
-Use this command if you want to set up the environment without actually running
+This command sets up the environment without actually running
```
Comment (dedemorton) [inline libbeat/docs/command-reference.asciidoc]: boom! so many extra words...thanks for cleaning this up.

File: libbeat/docs/outputconfig.asciidoc [hunk 5]
```diff
@@ -739,21 +739,39 @@ use in Logstash for indexing and filtering:
...
+<4> The current version of {beatname_uc}.
+endif::[]
 
 You can access this metadata from within the Logstash config file to set values
 dynamically based on the contents of the metadata.
 
-For example, the following Logstash configuration file for versions 2.x and
-5.x sets Logstash to use the index and document type reported by Beats for
-indexing events into Elasticsearch:
+For example, the following Logstash configuration file for version 7.x sets
```
Comment (dedemorton) [inline libbeat/docs/outputconfig.asciidoc]: Ouch...that should have been updated before now. Maybe we should not give a version at all here.

Comment (bmorelli25) [reply libbeat/docs/outputconfig.asciidoc]: Good idea. Removed.

**Comment by bmorelli25** [comment]
@dedemorton do you want me to partially backport this? The new Logstash config example would be useful to have in current documentation.

**Comment by dedemorton** [comment]
@bmorelli25  Sure, go for it!

**Pull request merged**